src/System.Diagnostics.PerformanceCounter/src/misc/EnvironmentHelpers.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Security;
+using System.Security.Principal;
+
+namespace System
+{
+    internal static class EnvironmentHelpers
+    {
+        private static volatile bool s_isAppContainerProcess;
+        private static volatile bool s_isAppContainerProcessInitalized;
+        internal const int TokenIsAppContainer = 29;
+
+        public static bool IsAppContainerProcess
+        {
+            get
+            {
+                if(!s_isAppContainerProcessInitalized)
+                {
+                   if(Environment.OSVersion.Platform != PlatformID.Win32NT)
+                   {
+                       s_isAppContainerProcess = false;
+                   }
+                   else if(Environment.OSVersion.Version.Major < 6 || (Environment.OSVersion.Version.Major == 6 && Environment.OSVersion.Version.Minor <= 1))
+                   {
+                       // Windows 7 or older.
+                       s_isAppContainerProcess = false;
+                   }
+                   else
+                   {
+                       s_isAppContainerProcess = HasAppContainerToken();
+                   }
+ 
+                    s_isAppContainerProcessInitalized = true;
+                }
+ 
+                return s_isAppContainerProcess;
+            }
+        }
+
+        [SecuritySafeCritical]
```

Comment:
you can remove all securitysafecritical and other security transparency attributes from the implementation, as it will only run on .NEt Core, which ignores them.